pkg/codegen/pcl/binder_resource.go

```diff
@@ -59,19 +59,29 @@ func (b *binder) bindResourceTypes(node *Resource) hcl.Diagnostics {
 		pkg, isProvider = name, true
 	}
 
-	pkgSchema, ok := b.options.packageCache.entries[pkg]
-	if !ok {
-		return hcl.Diagnostics{unknownPackage(pkg, tokenRange)}
+	var pkgSchema *packageSchema
+	if pkg != "pulumi" && module != "pulumi" {
+		var ok bool
+		pkgSchema, ok = b.options.packageCache.entries[pkg]
+		if !ok {
+			return hcl.Diagnostics{unknownPackage(pkg, tokenRange)}
+		}
+	} else {
+		isProvider = true
```

Comment:
Is is possible to assign `pkgSchema` a value here so we can prevent the conditional on line 76?